kernel/kobject.c

```diff
@@ -294,12 +316,7 @@ int kobject_close(struct kobject *kobject)
 
 int kobject_set_blocking(struct kobject *kobject, int b)
 {
-	switch (kobject->type) {
-	case KOBJECT_PIPE:
-		return pipe_set_blocking(kobject->data.pipe, b);
-	default:
-		return 0;
-	}
+	kobject->blocking = b;
```

Comment:
This looks like an incomplete solution to the blocking problem.
For the purpose of this PR, just leave blocking the way it is,
and fix the problem thoroughly in a different PR.